services/shhext/api_post.go

```diff
@@ -0,0 +1,42 @@
+package shhext
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+// Post shamelessly copied from whisper codebase with slight modifications.
+func (api *PublicAPI) Post(ctx context.Context, req whisper.NewMessage) (hash common.Hash, err error) {
```

Comment:
this will be simplified greatly after rebase. pleas consider having 2 methods in the same file (`api.go`)

Reply:
Actually, we talked about it to split it because in `api.go` we will have unrelated `Post` and mailservice methods.

Reply:
they both are extension over whisper, so in that sense, they are related)
but tbh post is only 4 lines now, and it will be a bit weird to have it in a separate file.
also @mandrigin works on one more extension for filters, where that should go in this case?

Reply:
Surely can move all back into `api.go`, also have no troubles with small files. 😁

Will move them back together in one file. As long as it's only so few code it's fine. Can think about new grouping approach once API grows too much.